### DISCUSSION TITLE
Feature Request: json format for EXPLAIN as well as for EXPLAIN ANALYZE

### DISCUSSION THREAD

**Discussion by jgm-ktg** [duckdb/duckdb#7575] [Ideas]
Programatic analysis of the query plan for unsliced data can help optimally slice data prior to the actual query run.

For example if an Arrow table is being queried and the query plan indicates a Filter where the caller knows that all such records are in a certain table row range (e.g. due to known sorting), this pre-slicing can avoid applying the columnar filter to every row.

This request for a json format for EXPLAIN as well as for EXPLAIN ANALYZE was also made (late) in ticket #3094.  As in that ticket, retrieving the json without going by way of a file would be ideal.

**Comment by jgm-ktg** [comment]
@bleskes @hannes @Mytherin thoughts? Would this be difficult?  It seems odd that EXPLAIN ANALYZE can be requested in json but not EXPLAIN...

**Comment by jgm-ktg** [comment]
In addition, the human-readable EXPLAIN output is sometimes truncated.  This would provide a complete view, it seems?

**Comment by Mytherin** [comment]
This is available now:

```sql
explain (format json) select 42;

┌─────────────────────────────┐
│┌───────────────────────────┐│
││       Physical Plan       ││
│└───────────────────────────┘│
└─────────────────────────────┘
[
    {
        "name": "PROJECTION",
        "children": [
            {
                "name": "DUMMY_SCAN",
                "children": [],
                "extra_info": {}
            }
        ],
        "extra_info": {
            "Projections": "42",
            "Estimated Cardinality": "1"
        }
    }
]
```
